<!-- Reusable launch file for obstacle detection -->
<launch>

  <node name="obstacle_extractor" pkg="obstacle_detector" exec="obstacle_extractor_node">
    <remap from="/scan" to="/merged_scan" />
    <remap from="/pcl" to="/scan_filtered" />
    <remap from="/pcl2" to="/ScanMergeNode/scan_merged" />
    <param name="active"               value="true"/>
    <param name="use_scan"             value="false"/>
    <param name="use_pcl"              value="false"/>
    <param name="use_pcl2"             value="true"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="10"/>

    <param name="max_group_distance"   value="0.1"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.6"/>
    <param name="radius_enlargement"   value="0.3"/>

    <param name="frame_id"             value="map"/>
  </node>

  <node name="obstacle_tracker" pkg="obstacle_detector" exec="obstacle_tracker_node">
    <remap from="/odom" to="/odometry/filtered" />

    <param name="active"                  value="true"/>

    <param name="loop_rate"               value="100.0"/>
    <param name="tracking_duration"       value="2.0"/>
    <param name="min_correspondence_cost" value="0.6"/>
    <param name="std_correspondence_dev"  value="0.15"/>
    <param name="process_variance"        value="0.1"/>
    <param name="process_rate_variance"   value="0.1"/>
    <param name="measurement_variance"    value="1.0"/>

    <param name="frame_id"                value="map"/>

    <remap from="tracked_obstacles" to="obstacles"/>
    <remap from="tracked_obstacles_visualization" to="obstacles_visualization"/>

  </node>

</launch>
<!-- -->
